docs/templates.md

```diff
@@ -0,0 +1,84 @@
+# Templates
+
+Often users will use the same templates. This documents describes the flow and
+APIs that will eventually come into place.
+
+## Template location
+
+There are two kind of templates:
+
+1. system wide templates
+2. user defined templates
+
+Both should be simple template files and preview for them should be obtained using
+the preview API. The richdocuments works as a preview provider so it should provide
+proper previews to use.
+
+### System wide templates
+
+An admin can configure system wide templates. This can be done from the admin
+settings.
+
+These templates are stored in the AppData.
+
+> Note: it might require some less pretty code to get the previews out of the
+appdata. But since it will also work on 13 we have no real alternative.
+
+There are a few special templates that are shipped with the app. The empty
+files. They should always be available.
+
+### User defined templates
+
+In the Collabora user settings the user can select a folder from where to load
+templates. This can be any folder the user has access to. So users can share template
+folders among each other.
+
+## API
+
+### Listing templates
+
+Send a GET to:
+
+`<server>/ocs/v2.php/apps/richdocuments/api/v1/templates/<type>`
+
+Here type can be:
+
+* document
+* spreadsheet
+* presentation
+
+This returns a list of templates with
+
+* id
+* name
+* link to preview
+* extension
+
+### Template previews
```

Comment:
I am not sure if we need this.
In my current idea I will retrieve the preview by the link above.